src/main/java/core/basesyntax/strategy/PurchaseOperation.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+    private final Storage storage;
+
+    public PurchaseOperation(Storage storage) {
+        this.storage = storage;
+    }
+
+    @Override
+    public void count(FruitTransaction transaction) {
+        if (transaction.getQuantity() < 0) {
+            throw new RuntimeException("Quantity can't be less than 0");
+        }
+        storage.removeFruit(transaction.getFruit(), transaction.getQuantity());

```

Comment:
The `removeFruit` method should handle the case where there is not enough fruit in the Storage to fulfill the purchase. Ensure that there is a check for sufficient stock before attempting to remove the fruit, and throw an informative RuntimeException if there isn't enough stock.